CMakeLists.txt

```diff
@@ -259,27 +239,39 @@ endif()
 
 
 # external library: nlohmann-json (required)
-if(openPMD_USE_INTERNAL_JSON)
-    set(JSON_BuildTests OFF CACHE INTERNAL "")
-    set(JSON_Install OFF CACHE INTERNAL "")  # only used PRIVATE
-    add_subdirectory("${openPMD_SOURCE_DIR}/share/openPMD/thirdParty/json")
-    message(STATUS "nlohmann-json: Using INTERNAL version '3.9.1'")
-else()
-    find_package(nlohmann_json 3.9.1 CONFIG REQUIRED)
-    message(STATUS "nlohmann-json: Found version '${nlohmann_json_VERSION}'")
+find_package(nlohmann_json CONFIG)
+if(NOT nlohmann_json_FOUND)
+    message(STATUS "Fetching nlohmann-json from https://github.com/nlohmann/json")
+    FetchContent_Declare(
+      nlohmann_json
+      GIT_REPOSITORY https://github.com/nlohmann/json.git
+      GIT_TAG v3.11.3
+      SOURCE_DIR "${openPMD_SOURCE_DIR}/share/openPMD/thirdParty/json/"
+      OVERRIDE_FIND_PACKAGE
+    )
+    set(JSON_BuildTests OFF CACHE INTERNAL "NLohmann JSON option defiend internally by OpenPMD")
+    set(JSON_Install OFF CACHE INTERNAL "NLohmann JSON option defiend internally by OpenPMD")  # only used PRIVATE
+    FetchContent_MakeAvailable(nlohmann_json)
 endif()
 add_library(openPMD::thirdparty::nlohmann_json INTERFACE IMPORTED)
 target_link_libraries(openPMD::thirdparty::nlohmann_json
     INTERFACE nlohmann_json::nlohmann_json)
 
 # external library: toml11
-if(openPMD_USE_INTERNAL_TOML11)
-    set(toml11_INSTALL OFF CACHE INTERNAL "")
-    add_subdirectory("${openPMD_SOURCE_DIR}/share/openPMD/thirdParty/toml11")
-    message(STATUS "toml11: Using INTERNAL version '3.7.1'")
-else()
-    find_package(toml11 3.7.1 CONFIG REQUIRED)
-    message(STATUS "toml11: Found version '${toml11_VERSION}'")
+find_package(toml11 CONFIG)
+if(NOT toml11_FOUND)
+    message(STATUS "Fetching toml11 from https://github.com/ToruNiina/toml11")
+    FetchContent_Declare(
+      toml11
+      GIT_REPOSITORY https://github.com/ToruNiina/toml11
+      # Migrate to the latest commit to remove CMake Warning which is not yet
+      # available in any official release.
```

Comment:
You mean the warning below? Do I understand correctly that this has been fixed on toml11 dev and we are waiting for the next release?

```
CMake Deprecation Warning at share/openPMD/thirdParty/toml11/CMakeLists.txt:1 (cmake_minimum_required):                                                                                                                    
  Compatibility with CMake < 3.5 will be removed from a future version of                                                                                                                                                  
  CMake.                                                                                                                                                                                                                   
                                                                                                                                                                                                                           
  Update the VERSION argument <min> value or use a ...<max> suffix to tell                                                                                                                                                 
  CMake that the project does not need compatibility with older versions. 
```

Reply:
Yes

Reply:
> since that requires `CMAKE_CXX_STANDARD` to be set which I could not add to openPMD-api w/o breaking the CI.

That's probably an issue unrelated to this PR which we should better solve separately?

Reply:
Yes, after finishing this PR I would try to Update catch2 and toml11 with separate issues as described above